packages/smooth_app/lib/pages/product/common/product_app_bar.dart

```diff
@@ -0,0 +1,92 @@
+import 'package:auto_size_text/auto_size_text.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:smooth_app/generic_lib/widgets/smooth_back_button.dart';
+import 'package:smooth_app/widgets/smooth_app_bar.dart';
+import 'package:smooth_app/widgets/smooth_floating_message.dart';
+
+class ProductAppBar extends StatefulWidget implements PreferredSizeWidget {
+  const ProductAppBar(
+      {required this.barcodeVisibleInAppbar,
+      required this.productName,
+      required this.productBrand,
+      required this.barcode});
+
+  final String productName;
+  final String productBrand;
+  final String barcode;
```

Comment:
Please just put the product itself as parameter, and not 3 parameters derived from the product.